### PR TITLE
@outsmartly/cli: add support for .ts(x?) files

### PR DESCRIPTION
**Pull request by kaicataldo**
Looks like the package-lock.json file is out of date (other random changes are from Prettier). I haven't been able to really test this yet, but I _think_ this should fix the issue we encountered.

@twop @eggers I’m out Monday, so feel free to test this out and merge it if fixes your issue!

### CODE REVIEW

File: packages/cli/src/commands/deploy.ts [hunk 1]
```diff
@@ -26,17 +19,11 @@ import { AbortSignal } from 'node-fetch/externals';
...
-  APIError,
-  apiFetch,
-  ComponentAnalysis,
-  patchSite,
-  PatchSite,
-} from '../api';
+import { Analysis, APIError, apiFetch, ComponentAnalysis, patchSite, PatchSite } from '../api';
 import { panic } from '../panic';
 
+const SUPPORTED_EXTENSIONS = ['.js', '.jsx', '.mjs', '.cjs', '.ts', '.tsx'];
```
Comment (kaicataldo) [inline packages/cli/src/commands/deploy.ts:25]: This [drops support for `.es` and `.es6`](https://github.com/rollup/plugins/tree/master/packages/babel#extensions). Does that seem reasonable, or do we want to support those extensions?

Comment (jayphelps) [reply packages/cli/src/commands/deploy.ts:25]: I think it's fine for now, long run we'll want to figure out some way of not caring about file extension at all, but I imagine that's what the webpack plugin will help us get to?

**Review by jayphelps — APPROVED** [review]
LGTM assuming it works as I have not tested.

File: packages/cli/src/commands/deploy.ts (hunk 1 above)
Comment (kaicataldo) [reply packages/cli/src/commands/deploy.ts:25]: Should, yeah!

**Comment by jayphelps** [comment]
Sorry for making the merge conflict 😢 hopefully it's easy. This should be ready to merge otherwise since we tested it.

**Review by twop — APPROVED** [review]
Verified

**Comment by kaicataldo** [comment]
Sorry, rebased and it dismissed the reviews 😢

**Review by twop — APPROVED** [review]
approved v2

**Pull request merged**